pkg/apis/serving/metadata_validation.go

```diff
@@ -41,9 +41,16 @@ var (
 
 // ValidateObjectMetadata validates that `metadata` stanza of the
 // resources is correct.
-func ValidateObjectMetadata(meta metav1.Object) *apis.FieldError {
+func ValidateObjectMetadata(ctx context.Context, meta metav1.Object) *apis.FieldError {
+	allowZeroInitialScale := false
+	if config.FromContext(ctx) != nil {
+		autoscalerConfig := config.FromContext(ctx).Autoscaler
+		if autoscalerConfig != nil {
```

Comment:
Why would this ever be nil? Just for tests?

Reply:
Yes this is just for the tests.

Reply:
Probably annotate tests instead? If we have cruft, I'd rather have it in tests than in production code.